python_modules/dagster/dagster/_core/definitions/asset_graph_differ.py

```diff
@@ -28,6 +27,79 @@ class AssetDefinitionChangeType(Enum):
     REMOVED = "REMOVED"
 
 
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailNew:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.NEW
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailCodeVersion:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.CODE_VERSION
+    base_value: Optional[str] = None
+    branch_value: Optional[str] = None
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailDependencies:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.DEPENDENCIES
+    added: Sequence[AssetKey] = []
```

Comment:
Why `Sequence` instead of `AbstractSet`? The values are expected to be unique, right?

Reply:
I'm using `Sequence` because eventually they need to be lists in GraphQL. I guess I can delay converting `AbsractSet` to `Sequence` until we need to?